### ISSUE TITLE
Allow trailing comments, print warning instead of dying hard for values with spaces

### ISSUE THREAD

**Issue by bittner** [vlucas/phpdotenv#236]
Trailing comments should be allowed in INI-style files, e.g.
``` ini
SENTRY_DSN=  # see https://docs.sentry.io/quickstart/#about-the-dsn
USE_COLOR=yes  # (yes, no, redish, blueish, transparent)
```

Don't Die Hard
-----------------

When I add a trailing comment in a `.env` file (instead of a full comment line) phpdotenv dies hard with this error:
``` php
Uncaught Dotenv\\Exception\\InvalidFileException: Dotenv values containing spaces must be surrounded by quotes.
```

In addition, parsing the `.env` file usually happens in a critical moment, at application startup time. This is critical especially when your error logging depends on the values in the file, for example. If this fails due to a simple change in the `.env` file the application crashes (Error 500), and you need to go find out what happened (without modern error reporting support).

Hence, instead of dying hard [`load()`](https://github.com/vlucas/phpdotenv/blob/master/src/Dotenv.php#L46) should rather print a warning than tearing the application down. Reading environment values should be robust, not fragile.

Additional Details
--------------------------

To be fair I need to mention that support for trailing comments seems to be debated. According to [a StackOverflow answer](https://stackoverflow.com/questions/1378219/do-standard-windows-ini-files-allow-comments) on Windows this seems not to be supported, whereas some implementations of INI file format (Cloanto) allow this.

The Python implementation, for example, allows inline comments if the value is non-empty (not otherwise, according to the documentation [quoted in another SO answer](https://stackoverflow.com/questions/9492430/inline-comments-for-configparser)).

Related Issues
---------------

- #232 (Q: Not set Value)
- #170 (Cannot set path with spaces)
- #133 (Why is it required to have envvars with spaces to be surrounded by quotes?)

**Comment by bittner** [comment]
Seems like trailing comments _with_ a value already work:
``` ini
SENTRY_DSN=''  # empty value; this does not throw an exception
```

Robustness with empty values, and with spaces, would still be good, though.

**Comment by GrahamCampbell** [comment]
Fixed in 2.5.0. :)
